exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -218,13 +218,18 @@ private void javaPropertiesToSystemOptions() {
   }
 
   /**
-   * If cluster version is the same as default, update option to current drillbit version.
+   * If cluster version is the same as default, updates option to current drillbit version.
+   * Since getOption and setOption are separate calls, raise condition might occur
+   * when several drillbits are registering at the same time.
+   * It is assumed that the first drillbit in cluster sets cluster version
+   * but when the raise condition occurs, it might be the second or the third etc.
+   * This behaviour does not impose significant impact, since the main goal is to detect mismatching versions.
    */
-  private void checkAndUpdateClusterVersionOption(OptionManager optionManager, DrillbitEndpoint drillbitEndpoint) {
-    OptionValue versionOption = optionManager.getOption(ExecConstants.CLUSTER_VERSION);
+  private void checkAndUpdateClusterVersionOption(OptionManager systemOptions, DrillbitEndpoint drillbitEndpoint) {
```

Comment:
Actually, I don't think this master version is even needed. Instead:

1. Each time a new Drillbit is registered from ZK (see ZKClusterCoordinator),
2. Check the version of that new Drillbit against the version of this Drillbit.
3. If mismatch, log an error.
4. In the web UI, display the version of this Drillbit.
5. In the list of Drillbits, highlight any Drillbit with a different version.
6. If any Drillbits have a different version, display a warning message after the list of Drillbits.

With the above approach, there is no state to be cleared if the admin corrects the problem. With the cluster version state, it may be that someone has to reset the old cluster version when the mismatch problem is fixed.